### PR TITLE
RD-5773 - Migration of inputs widget

### PR DESCRIPTION
**Pull request by Blewin**
## Description

migrating of nputs widget

## Screenshots / Videos

no visual changes intended
## Checklist

- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests

 

typechecks should be my tests
## Documentation

N/a

### CODE REVIEW

File: widgets/inputs/src/InputsTable.tsx [hunk 1]
```diff
@@ -1,7 +1,23 @@
-// @ts-nocheck File not migrated fully to TS
+export interface InputsTableProps {
+    data: {
+        blueprintId: string | string[] | null | undefined;
+        deploymentId: string | string[] | null | undefined;
+        items: {
+            description: string;
+            name: string;
+            value: unknown; // Stage.PropTypes.anydata
```
Comment (Blewin) [inline widgets/inputs/src/InputsTable.tsx:8]: is unknown ok here?

File: widgets/inputs/src/InputsTable.tsx [hunk 2]
```diff
@@ -1,7 +1,23 @@
...
+        }[];
+    };
+    toolbox: Stage.Types.Toolbox;
+    widget: Stage.Types.Widget;
+}
+
+export interface InputsTableState {
+    error: null; // Error is never different from null in this widget
+}
```
Comment (qooban) [inline widgets/inputs/src/InputsTable.tsx:17]: If `error` is always set to `null`, then let's remove it from state.

File: widgets/inputs/src/widget.tsx [hunk 3]
```diff
@@ -1,8 +1,11 @@
...
 
-Stage.defineWidget({
+export interface InputsTableData {
+    inputs: InputsTableProps['data']['items'];
+}
+
+Stage.defineWidget<unknown, InputsTableData, Record<string, unknown>>({
     id: 'inputs',
     name: 'Deployment Inputs',
     description: 'This widget shows the deployment inputs',
```
Comment (qooban) [inline widgets/inputs/src/widget.tsx:11]: Let's add translation support to this widget at once.
Applies to both files.

(during TS migration it's not mandatory to do so, but it's strongly recommended - RD-5588)

File: widgets/inputs/src/InputsTable.tsx (hunk 1 above)
Comment (qooban) [reply widgets/inputs/src/InputsTable.tsx:8]: Yeah, I think it's OK.

File: widgets/inputs/src/widget.tsx [hunk 4]
```diff
@@ -1,8 +1,11 @@
...
-
+import type { InputsTableProps } from './InputsTable';
 import InputsTable from './InputsTable';
 
-Stage.defineWidget({
+export interface InputsTableData {
+    inputs: InputsTableProps['data']['items'];
+}
+
+Stage.defineWidget<unknown, InputsTableData, Record<string, unknown>>({
```
Comment (qooban) [inline widgets/inputs/src/widget.tsx:8]: Regarding parameters:
1. `unknown` - it's fine.
2. `InputsTableData` - the type seems correct, however - OPTIONAL: I'm wondering wouldn't it be more logically to extract common part of it and reuse in both files? 🤔 (in some widgets we used that concept of having `widget.types.ts` file)
3. `Record<string, unknown>` - we can be more specific here as we know exctly what configuration this widget exposes (see `initialConfiguration` property - `Stage.GenericConfig.POLLING_TIME_CONFIG(30)`). Probably you can have here `PollingTimeConfiguration` (from `app/utils/GenericConfig.ts`).

Comment (Blewin) [reply widgets/inputs/src/widget.tsx:8]: ad 2: good idea, can i just add InputItem interface to inputs table, or do you think new file is necessary

Comment (Blewin) [reply widgets/inputs/src/widget.tsx:8]: 3 done

File: app/translations/en.json [hunk 5]
```diff
@@ -1695,6 +1695,15 @@
                 }
             }
         },
+        "inputs": {
+            "name": "Deployment Inputs",
+            "description": "This widget shows the deployment inputs",
+            "noData": "There are no Inputs available. Probably there's no deployment created, yet.",
+            "columnLabels": {
```
Comment (qooban) [inline app/translations/en.json:1702]: Can we stick to the convention in this file and name it `columns`?

File: widgets/inputs/src/widget.tsx [hunk 6]
```diff
@@ -1,11 +1,16 @@
...
-Stage.defineWidget({
+export interface InputsTableData {
+    inputs: InputItem[];
+}
+
+Stage.defineWidget<unknown, InputsTableData, PollingTimeConfiguration>({
     id: 'inputs',
-    name: 'Deployment Inputs',
-    description: 'This widget shows the deployment inputs',
+    name: Stage.i18n.t('widgets.inputs.name'),
```
Comment (qooban) [inline widgets/inputs/src/widget.tsx:12]: What do you think about extracting common function for translation in this widget?

You can get inspired with these examples:

image

BTW We've got some general guidelines for the translation here: [UI Translation Files Guidelines](https://cloudifysource.atlassian.net/wiki/spaces/RD/pages/1875083446/UI+Translation+Files+Guidelines)
(it's referenced from UI Code Style page)

File: widgets/inputs/src/widget.tsx (hunk 4 above)
Comment (qooban) [reply widgets/inputs/src/widget.tsx]: It looks good! Thanks!

**Review by qooban — COMMENTED** [review]
Thanks for addressing my comments. 

Looks much better right now IMHO and we're getting "there" 🙂 
I added two more comments, I believe it won't be painful to address. Let me know otherwise.

File: widgets/inputs/src/widget.tsx (hunk 6 above)
Comment (Blewin) [reply widgets/inputs/src/widget.tsx:12]: I dislike one letter names, but i guess its a convention in this repo at this point.

(as for extracting, extracting for 3 occurences seems to be on the border, so lets do it :)

Comment (Blewin) [reply widgets/inputs/src/widget.tsx]: but when i wanted to also export it, i couldnt bear to name it "T"

**Review by qooban — APPROVED** [review]
Let's wait until we have final agreement on translation functions naming, but in general: Looks good 👍🏻

**Pull request merged**